pkg/monitor/controller_test.go

```diff
@@ -0,0 +1,719 @@
+		})
+		controller := newControllerMonitor(clientset)
+		stopCh := make(chan struct{})
+		initController(controller, stopCh)
+		defer close(stopCh)
+		controller.controllerMonitor.syncControllerCRD()
+		assert.Nil(t, controller.controllerMonitor.controllerCRD)
+	})
+	t.Run("create-success", func(t *testing.T) {
+		clientset := fakeclientset.NewSimpleClientset()
+		controller := newControllerMonitor(clientset)
+		stopCh := make(chan struct{})
+		initController(controller, stopCh)
+		defer close(stopCh)
+		controller.controllerMonitor.querier.GetControllerInfo(newCRD, false)
+		controller.controllerMonitor.syncControllerCRD()
+		crd, err := controller.controllerMonitor.client.CrdV1beta1().AntreaControllerInfos().Get(ctx, crdName, metav1.GetOptions{})
+		newCRD.ControllerConditions[0].LastHeartbeatTime.Time = crd.ControllerConditions[0].LastHeartbeatTime.Time
+		require.NoError(t, err)
+		assert.Equal(t, newCRD, crd)
+	})
+	t.Run("create-failure", func(t *testing.T) {
+		clientset := fakeclientset.NewSimpleClientset()
+		clientset.PrependReactor("create", "antreacontrollerinfos", func(action cgtesting.Action) (handled bool, ret runtime.Object, err error) {
+			return true, &v1beta1.AntreaControllerInfo{}, errors.New("error in creating controller crd")
+		})
+		controller := newControllerMonitor(clientset)
+		stopCh := make(chan struct{})
+		initController(controller, stopCh)
+		defer close(stopCh)
+		controller.controllerMonitor.syncControllerCRD()
+		assert.Nil(t, controller.controllerMonitor.controllerCRD)
+	})
+	t.Run("entire update-success", func(t *testing.T) {
+		clientset := fakeclientset.NewSimpleClientset(existingCRD)
+		controller := newControllerMonitor(clientset)
+		stopCh := make(chan struct{})
+		initController(controller, stopCh)
+		defer close(stopCh)
+		controller.controllerMonitor.querier.GetControllerInfo(entirelyUpdatedCRD, false)
+		controller.controllerMonitor.syncControllerCRD()
+		crd, err := controller.controllerMonitor.client.CrdV1beta1().AntreaControllerInfos().Get(ctx, crdName, metav1.GetOptions{})
+		assert.NoError(t, err)
+		entirelyUpdatedCRD.ControllerConditions[0].LastHeartbeatTime.Time = crd.ControllerConditions[0].LastHeartbeatTime.Time
+		assert.Equal(t, entirelyUpdatedCRD, crd)
+	})
+	t.Run("entire update-failure", func(t *testing.T) {
+		clientset := fakeclientset.NewSimpleClientset(existingCRD)
+		clientset.PrependReactor("update", "antreacontrollerinfos", func(action cgtesting.Action) (handled bool, ret runtime.Object, err error) {
+			return true, &v1beta1.AntreaControllerInfo{}, errors.New("error updating controller crd")
+		})
+		controller := newControllerMonitor(clientset)
+		stopCh := make(chan struct{})
+		initController(controller, stopCh)
+		defer close(stopCh)
+		controller.controllerMonitor.syncControllerCRD()
+		assert.Nil(t, controller.controllerMonitor.controllerCRD)
+	})
+}
+
+func TestEnqueueNode(t *testing.T) {
```

Comment:
I don't think these "Enqueue" tests are very useful, this is what the function is like:
```go
func (monitor *controllerMonitor) enqueueNode(obj interface{}) {
	node := obj.(*corev1.Node)
	key, _ := keyFunc(node)
	monitor.nodeQueue.Add(key)
}
```

And if you really want to write a test for it, it can be much shorter. I don't think you need to call `initController` for example.

Reply:
I have removed init Controller from the two test cases ....Do these need to be shortened further?These functions are not getting covered by other functions , that is why I have included these tests.